### DISCUSSION TITLE
jsii warning in regards to deps, devDeps, and peerDeps

### DISCUSSION THREAD

**Discussion by anh-rivergon** [projen/projen#2207] [Q&A]
I have the following dependencies in .projenrc.js 
```ts
  deps: [
    '@aws-solutions-constructs/aws-cloudfront-s3',
    '@aws-solutions-constructs/core',
  ], /* Runtime dependencies of this module. */
  devDeps: [
    '@aws-solutions-constructs/aws-cloudfront-s3',
  ], /* Build dependencies for this module. */
  peerDeps: [
    '@aws-solutions-constructs/aws-cloudfront-s3',
  ],
```
Yet still, here is this warning below. Changing devDeps to devDependency and devDependencies did not eliminate the warning.
What is the right way to declare dependency on @aws-solutions-constructs/aws-cloudfront-s3 ?
```
👾 build » default | node .projenrc.js
👾 build » compile | jsii --silence-warnings=reserved-word
warning JSII6: A "peerDependency" on "@aws-solutions-constructs/aws-cloudfront-s3" at "^2.25.0" means you should take a "devDependency" on "@aws-solutions-constructs/aws-cloudfront-s3" at "2.25.0" (found "^2.25.0")
```

**Comment by mrgrain** [comment]
Hi @anh-rivergon if the runtime code of your package depends on `@aws-solutions-constructs/aws-cloudfront-s3` you should define it as dependency using `deps`. The jsii warning is just that, a warning and you might choose to ignore it without much harm.

--- 

If you rather fix the warning:

Jsii stipulates that you should test against the lowest available version matching the version constraint of any dependency. In Projen, when you do not specify a version constraint, Projen will infer the version constraint to use. In your case that appears to be `^2.25.0`.  Hence why jsii warns you that depending on "^2.25.0"`  means you should also take a dev dependency on "2.25.0"`.  The way to get around this would be to specify concrete version constraints:
```ts
deps: [
  '@aws-solutions-constructs/aws-cloudfront-s3@^2.25.0',
],
devDeps: [
  '@aws-solutions-constructs/aws-cloudfront-s3@2.25.0',
],
```
